Assignment 2/Pop Alexia/Assignment2SD/backend/src/main/java/com/lab4/demo/book/BookController.java

```
@@ -0,0 +1,78 @@
+package com.lab4.demo.book;
+
+import com.lab4.demo.book.model.dto.BookDTO;
+import com.lab4.demo.report.ReportType;
+import com.lab4.demo.security.dto.MessageResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.ConstraintViolationException;
+import javax.validation.Valid;
+import java.util.List;
+import java.util.Set;
+
+import static com.lab4.demo.UrlMapping.*;
+
+@RestController
+@RequestMapping(BOOK)
+@RequiredArgsConstructor
+public class BookController {
+
+    private final BookService bookService;
+
+
+    @GetMapping
+    public List<BookDTO> allItems() {
+        return bookService.findAll();
+    }
+
+    @GetMapping("/filter/{filter}")
+    public List<BookDTO> filterBooks(@PathVariable String filter){return bookService.filterBooks(filter);}
+
+    @PostMapping
+    public ResponseEntity<?> create(@Valid @RequestBody BookDTO book) {
+        if(bookService.findByTitle(book.getTitle())!=null){
```

First of all, this is not a check which you should be doing
- on a controller level
- at all, really

Why?

Because you are manually doing the database's work. Why not put a [unique constraint](https://www.baeldung.com/jpa-unique-constraints) on the title column and then let the database throw this exception when you'd be trying to create the entity?